app/src/common/shared/com/igalia/wolvic/ui/widgets/WindowWidget.java

```diff
@@ -1679,7 +1679,9 @@ public void onExternalResponse(@NonNull WSession aSession, @NonNull WWebResponse
         // so we let the system handle it.
         if (!UrlUtils.isFileUri(webResponseInfo.uri())) {
             DownloadJob job = DownloadJob.fromUri(webResponseInfo.uri());
-            startDownload(job, true);
+            // Don't show the confirmation dialog when we are in WebXR, because it will not be visible.
+            boolean showConfirmDialog = !mWidgetManager.isWebXRPresenting();
+            startDownload(job, showConfirmDialog);
```

Comment:
This is another example why bools in APIs is a bad idea. You're forced to create a variable just to make explicit what the boolean is about. Ideally we should have an enumerated type, something like
```
enum ShowConfirmDialog {
Yes,
No
}
```
and we could call the method with `startDownload(job, ShowConfirmationDialog:Yes)` for example.

Anyway I'm fine with your solution (you didn't design the API anyway), feel free to do the enum thing now or later at some point.